lib/index.js

```diff
@@ -85,17 +88,47 @@ FacebookPixel.prototype.page = function() {
 FacebookPixel.prototype.track = function(track) {
   var event = track.event();
   var revenue = formatRevenue(track.revenue());
-
   var payload = foldl(function(acc, val, key) {
     if (key === 'revenue') {
       acc.value = revenue;
       return acc;
     }
 
+    /**
+    * FB requires these date fields be formatted in a specific way.
+    * The specifications are non iso8601 compliant.
+    * https://developers.facebook.com/docs/marketing-api/dynamic-ads-for-travel/audience
+    * Therefore, we check if the property is one of these reserved fields.
+    * If so, we check if we have converted it to an iso date object already.
+    * If we have, we convert it again into Facebook's spec.
+    * If we have not, the user has likely passed in a date string that already
+    * adheres to FB's docs so we can just pass it through as is.
+    * @ccnixon
+    */
+
+    var dateFields = [
+      'checkinDate',
+      'checkoutDate',
+      'departingArrivalDate',
+      'departingDepartureDate',
+      'returningArrivalDate',
+      'returningDepartureDate',
+      'travelEnd',
+      'travelStart'
+    ];
+
+    if (dateFields.indexOf(camel(key)) >= 0) {
+      if (is.date(val)) {
+        val = val.toISOString().split('T')[0];
```

Comment:
this seems like a danger line of code, are we sure we dont have a chance of hitting uncaughts